### ISSUE TITLE
Widgets with custom UI go missing

### ISSUE THREAD

**Issue by simison** [WordPress/gutenberg#25836]
**Describe the bug**
[Jetpack plugin](https://github.com/Automattic/jetpack) has a "widget visibility" feature which extends capabilities of all the regular widgets.

When custom visibility has been set to a widget in the classic screen, the whole widget goes missing on the new screen. It still appears on the frontend of the site just fine.

**To reproduce**
1. Install Jetpack, and connect it to WordPress.com
2. Go to `/wp-admin/admin.php?page=jetpack#/writing` and enable widget visibility:

    <img width="505" alt="Screenshot 2020-10-05 at 16 40 55" src="https://user-images.githubusercontent.com/87168/95086843-9f252f80-0729-11eb-8bf5-64aca215c798.png">
3. Go to legacy widget screen (or customizer), and add some visibility setting to any core-widget from "visibility" button:

    <img width="908" alt="Screenshot 2020-10-05 at 15 31 01" src="https://user-images.githubusercontent.com/87168/95086925-ba903a80-0729-11eb-88d9-56e92c3128df.png">
4. Enable new Widgets screen and have a look — the widget is gone:

    <img width="908" alt="Screenshot 2020-10-05 at 15 41 25" src="https://user-images.githubusercontent.com/87168/95087050-e14e7100-0729-11eb-87cc-13ba972cd8cf.png">
5. Confirm that widget still appears on the site:

    <img width="408" alt="Screenshot 2020-10-05 at 15 41 25" src="https://user-images.githubusercontent.com/87168/95087129-f925f500-0729-11eb-87bd-5a1c14a65c8e.png">

**Expected behavior**
- I expected all the custom UI to go missing, no problem there. It needs to be re-implemented.
- I did expect the widget itself to appear on the list in the name of backward compatibility. If it cannot be shown as functional, I would've expected it to show up in the list with some "incompatible" indication.
- In other words, I expected all widgets that appear at the site, be also visible in some way or the other in the widgets screen.

**Editor version (please complete the following information):**
- WordPress version 5.5.1
- Gutenberg plugin 9.1.0

**Comment by adamziel** [comment]
Related: pbAPfg-Oa-p2

**Comment by adamziel** [comment]
Related: https://github.com/WordPress/gutenberg/issues/18068 https://github.com/WordPress/gutenberg/pull/25758

**Comment by simison** [comment]
I think the UI would appear at block toolbar or similar like you'd extend any old block in the page editor? That's secondary though. I think the backward compatibility issue at backend seems more important.

**Comment by noisysocks** [comment]
Hey @simison, I wasn't able to reproduce this with the latest Gutenberg `master`.

Here's a GIF demonstrating my testing steps:

![jetpack](https://user-images.githubusercontent.com/612155/95544303-701efe80-0a46-11eb-9c94-c0ab43d24367.gif)

Let me know if I've made a mistake or if you're still encountering this and we can re-open.

**Comment by draganescu** [comment]
Normally all the form options should appear. The bug here is the widget is completely gone, but in general a legacy widget shown by the Legacy Widget block in the Widget editor should show the exact same edit form it did in the old editor (screen).

To update the UI and use block extensibility or editor extensibility options, the widget has to be "migrated" (reimplemented) as a block and the legacy widget will not be used automatically.

**Comment by adamziel** [comment]
I managed to reproduce the problem on Gutenberg 9.1.1:

<img width="489" alt="Zrzut ekranu 2020-10-9 o 16 12 32" src="https://user-images.githubusercontent.com/205419/95593659-436ce600-0a4a-11eb-9905-3a07797bacb4.png">
<img width="770" alt="Zrzut ekranu 2020-10-9 o 16 12 36" src="https://user-images.githubusercontent.com/205419/95593673-4667d680-0a4a-11eb-8d90-3587d0b31815.png">

That being said, things are back to normal immediately after switching back to `master`:

<img width="737" alt="Zrzut ekranu 2020-10-9 o 16 14 16" src="https://user-images.githubusercontent.com/205419/95593901-862ebe00-0a4a-11eb-84c6-f39cd78ae58c.png">

It seems like one of the PRs we merged in the meantime fixed the problem.
